src/gwc/src/main/java/org/geoserver/gwc/GWC.java

```diff
@@ -2187,6 +2188,34 @@ public void verifyAccessLayer(String layerName, ReferencedEnvelope boundingBox)
             }
         }
     }
+    
+    /**
+     * Verify that a layer is accessible within a certain bounding box (calculated from tile) using the (secured) catalog
+     * 
+     * @param layerName name of the layer
+     * @param srs reference system
+     * @param level zoom level
+     * @param tileColumn column of tile in tile grid
+     * @param tileRow row  of tile in tile grid
+     * @throws ServiceException
+     */
+    public void verifyAccessTiledLayer(String layerName, String srs, int level, long tileColumn,
+            long tileRow) throws ServiceException {
+        // get bounding box of requested tile
+        GeoServerTileLayer layer = (GeoServerTileLayer) getTileLayerByName(layerName);
+        GridSubset gridSubset = layer.getGridSubset(srs);
```

Comment:
Gridsets, and coordinate systems are different things, do not confuse them.  I think you should be calling this parameter "gridset" or "gridset name"  A CRS can have any number of gridsets using it.  GeoWebCache, for legacy reasons, has several gridsets that are simply named for the CRSes they use, but this is not necessarily the case.